exporters/memory/src/in_memory_span_data.cc

```diff
@@ -0,0 +1,42 @@
+#include "opentelemetry/exporters/memory/in_memory_span_data.h"
+
+using opentelemetry::sdk::common::CircularBuffer;
+using opentelemetry::sdk::common::CircularBufferRange;
+using opentelemetry::sdk::common::AtomicUniquePtr;
+
+const size_t MAX_BUFFER_SIZE = 100;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace memory
+{
+InMemorySpanData::InMemorySpanData()
+{
+  CircularBuffer<sdk::trace::SpanData> buffer{MAX_BUFFER_SIZE};
+  spans_received_ = buffer;
+}
+
+void InMemorySpanData::Add(std::unique_ptr<sdk::trace::SpanData> data) noexcept
+{
+	spans_received_.Add(data);
+}
+
+std::vector<sdk::trace::SpanData> InMemorySpanData::GetSpans() noexcept
+{
+	std::vector<sdk::trace::SpanData> res;
+
+	spans_received_.Consume(
+		spans_received_.size(), [&](CircularBufferRange<AtomicUniquePtr<sdk::trace::SpanData>> & range) noexcept {
+			range.ForEach([&](AtomicUniquePtr<sdk::trace::SpanData> & ptr) noexcept {
+				res.push_back(*ptr);
+				ptr.Reset();
```

Comment:
I think you have to use `Swap` here. That's a bit cumbersome, we should add something to the interfaced of `AtomicUniquePtr` to make this more convenient. For now, do something similar to [here](https://github.com/open-telemetry/opentelemetry-cpp/pull/195/files#diff-8f953be5a9c81b02484d4faea832a5f6R189).

Reply:
Updated